functional-tests/src/main/java/com/redhat/che/selenium/core/workspace/CheStarterWrapper.java

```diff
@@ -11,166 +11,209 @@
-    } catch (IOException e) {
-      LOG.error("Workspace could not be deleted : " + e.getMessage(), e);
+      response =
+          this.jsonRequestFactory
+              .fromUrl(this.cheStarterURL.concat(relativePath))
+              .setAuthorizationHeader(authorization)
+              .addQueryParams(getDefaultQueryParameters())
+              .setMethod(HttpMethod.DELETE)
+              .request();
+    } catch (NotFoundException e) {
+      LOG.warn("Workspace could not be deleted because workspace was not found.");
+      return true;
+    }
+    if (response.getResponseCode() == 200) {
+      LOG.info("Workspace {} deleted successfully.", workspaceName);
+      return true;
+    } else {
+      LOG.error("Workspace could not be deleted: " + response.asString());
       return false;
     }
   }
 
-  public void startWorkspace(String WorkspaceID, String name, String token) throws Exception {
-    OkHttpClient client = new OkHttpClient();
-    String path = "/workspace/" + name;
-    StringBuilder sb = new StringBuilder(this.cheStarterURL);
-    sb.append(path);
-    sb.append("?");
-    sb.append("masterUrl=").append(this.host).append("&").append("namespace=sthf");
-    Builder requestBuilder = new Request.Builder().url(sb.toString());
-    requestBuilder.addHeader("Authorization", "Bearer " + token);
-    RequestBody body = RequestBody.create(null, new byte[0]);
-    Request request = requestBuilder.patch(body).build();
-    try {
-      Response response = client.newCall(request).execute();
+  public void startWorkspace(String WorkspaceID, String workspaceName, String token)
+      throws Exception {
+    String relativePath = "/workspace/" + workspaceName;
+    String authorization = "Bearer " + token;
+    RequestBody body = RequestBody.create(MediaType.parse("application/json; charset=urf-8"), "{}");
+
+    StringBuilder cheServerURLBuilder = new StringBuilder(this.cheStarterURL);
+    cheServerURLBuilder.append(relativePath);
+    cheServerURLBuilder.append('?');
+    getDefaultQueryParameters()
+        .forEach(
+            (key, value) -> cheServerURLBuilder.append(key).append('=').append(value).append('&'));
+    cheServerURLBuilder.deleteCharAt(cheServerURLBuilder.lastIndexOf("&"));
+
+    Request request =
+        new Request.Builder()
+            .url(cheServerURLBuilder.toString())
+            .addHeader("Authorization", authorization)
+            .patch(body)
+            .build();
+
+    OkHttpClient client = new OkHttpClient.Builder().build();
+    Response response = client.newCall(request).execute();
+    if (response.isSuccessful()) {
+      relativePath = "workspace/" + WorkspaceID + "/runtime";
+      request =
+          new Request.Builder()
+              .url(testApiEndpointUrlProvider.get().toString().concat(relativePath))
+              .addHeader("Authorization", authorization)
+              .post(body)
+              .build();
+      response = client.newCall(request).execute();
```

Comment:
Is it needed to use OkHttpClient here? Wouldn't it be better to use jsonRequestFactory? I think OkHttpClient should be used only for PATCH request.